pkg/virt-controller/watch/pod.go

```diff
@@ -87,22 +87,42 @@ func NewPodControllerWithListWatch(vmCache cache.Indexer, recorder record.EventR
 			vmCopy.ObjectMeta.Labels[corev1.NodeNameLabel] = pod.Spec.NodeName
 			vmCopy.Status.NodeName = pod.Spec.NodeName
 			// Update the VM
-			logger := logging.DefaultLogger().Object(vm)
-			if err := restClient.Put().Resource("vms").Body(&vmCopy).Name(vmCopy.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error(); err != nil {
-				logger.Error().Reason(err).Msg("Setting the VM to pending failed.")
-				if e, ok := err.(*errors.StatusError); ok {
-					if e.Status().Reason == metav1.StatusReasonNotFound {
-						// VM does not exist anymore, we don't have to retry
-						return true
-					}
-
-				}
-				logger.V(3).Info().Msg("Enqueuing VM initialization again.")
-				queue.AddRateLimited(key)
+			logger := logging.DefaultLogger()
+			if putVm(vm, restClient, queue, key) {
+				return true
+			}
+			logger.Info().Msgf("VM successfully scheduled to %s.", vmCopy.Status.NodeName)
+		} else if vm.Status.Phase == corev1.Running {
+			vmCopy := corev1.VM{}
+			model.Copy(&vmCopy, vm)
```

Comment:
In the case of coping from one object to another object of the same type you can use the kubernetes copier `api.Scheme.Copy()`. It does the same thing but looks nicer ;)